vortex-array/src/compute/search_sorted.rs

```diff
@@ -23,26 +23,59 @@ impl Display for SearchSortedSide {
     }
 }
 
+/// Result of performing search_sorted on an Array
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub enum SearchResult {
     Found(usize),
     NotFound(usize),
 }
 
 impl SearchResult {
+    /// Convert search result to an index only if the value have been found
     pub fn to_found(self) -> Option<usize> {
         match self {
             Self::Found(i) => Some(i),
             Self::NotFound(_) => None,
         }
     }
 
+    /// Extract index out of search result regardless of whether the value have been found or not
     pub fn to_index(self) -> usize {
         match self {
             Self::Found(i) => i,
             Self::NotFound(i) => i,
         }
     }
+
+    /// Convert search result into an index suitable for index 0 offset ends arrays, like ChunkArray ends
```

Comment:
it's not immediately obvious what you mean by "index 0 offset ends arrays" without looking at the code you reference, maybe expand on the semantics here a bit?